pkg/services/hcloud/remediation/remediation.go

```diff
@@ -143,6 +149,7 @@ func (s *Service) handlePhaseWaiting(ctx context.Context) (res reconcile.Result,
 	s.scope.HCloudRemediation.Status.Phase = infrav1.PhaseDeleting
 
 	if err := s.setOwnerRemediatedCondition(ctx); err != nil {
+		record.Warn(s.scope.HCloudRemediation, "FailedSettingConditionOnMachine", err.Error())
```

Comment:
you can make another event below setOwnerRemediatedCondition that this condition was set now and we exit remediation. 

Please check again all occurrences of setOwnerRemediatedCondition in both controllers to make sure that you found all cases and set events